ansible/tasks/installOpenwhiskCatalog.yml

```diff
@@ -0,0 +1,22 @@
+---
+# This task will install the standard actions and packages available in openwhisk-catalog repos.
+
+- set_fact:
+    catalog_location={{ item.value.location }}
+    catalog_repo_url={{ item.value.url }}
+    auth_key="{{ lookup('file', '{{ auth_key_file }}') }}"
+
+- stat: path="{{ catalog_location }}"
+  register: catalog_path
+ 
+- name: download the catalog repository to the catalog location if necessary
+  git:
+    repo: "{{ catalog_repo_url }}"
+    dest: "{{ catalog_location }}"
+    depth: 1
+  when: catalog_path.stat.isdir is not defined
```

Comment:
Yes. It the catalog directory already exists, we will skip the process of downloading the catalog repository.